docs/WritingTests.asciidoc

```diff
@@ -528,20 +525,21 @@ sub run {
 ====
 
 
-Mutexes are a way to wait for specific event from single job.
-When we need multiple jobs to reach required state we need to use barriers.
-
-Before first use barrier needs to be created with barrier_create with 2 parameters - name and count.
-Name behaves as ID (same as with mutexes), count is number of jobs needed to call barrier_wait to unlock barrier.
+Mutexes are a way to wait for specific events from a single job.
+When we need multiple jobs to reach a certain state we need to use barriers.
 
-There is optional barrier_wait parameter called check_dead_job.
-When used it will kill all jobs waiting in barrier_wait if one of cluster jobs die.
+To create a barrier call +barrier_create+ with the parameter name and count.
+The name serves as an ID (same as with mutexes). The count parameter specifies the
+number of jobs needed to call +barrier_wait+ to unlock barrier.
 
-It prevents waiting for state that will never be reached (and eventually die on job timeout).
-Should be set only on one of barrier_wait calls.
+There is an optional +barrier_wait+ parameter called +check_dead_job+.
+When used it will kill all jobs waiting in +barrier_wait+ if one of the cluster jobs
+dies. It prevents waiting for states that will never be reached (and eventually dies
+on job timeout). It should be set only on one of +barrier_wait+ calls.
 
-Example is situation with 1 master and 3 worker jobs. We need to wait until 3 worker jobs perform initial setup.
-After that we can make cluster from them, but if one of them fails it makes no sense waiting.
+Example is a situation with 1 master and 3 worker jobs. We need to wait until 3 worker
```

Comment:
```
For example when you have 1 master and 3 worker jobs and you want to do initial setup in 3 worker jobs before starting main actions. In such case you might use +check_dead_job+ to avoid useless actions when one of worker jobs dies
```

"After that we can make a cluster from them" - hmm just few blocks before this you were speaking about "dependency clusters" now you saying that you can create cluster inside a test. I would understand that I can create "dependency cluster" inside a test , right ?

Reply:
I changed the "example sentence" similar to your suggestion. The "After that we can make a cluster from them" makes no sense to me at all. I've just added an article but I better remove it completely if the meaning is not clear. 

(So I don't know what the writer of the documentation meant with "cluster" here. We actually/usually use the term to speak about tightly coupled jobs in a dependency tree. Using it here in a different sense is causing confusion.)